engine/conftest.py

```diff
@@ -0,0 +1,349 @@
+# coding = utf-8
+# Create date: 2018-10-29
+# Author :Bowen Lee
+
+from __future__ import print_function
+
+import os
+import random
+import string
+import subprocess
+import time
+from xml.etree.ElementTree import ElementTree
+import libvirt
+import pexpect
+import pytest
+import paramiko
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+@pytest.fixture
+def ros_kvm():
+    dom = None
+    conn = None
+    virtual_name = None
+
+    def _ros_kvm(cloud_config):
+        nonlocal virtual_name
+        virtual_name = _id_generator()
+        mac = _mac_generator()
+
+        example = '''<domain type='kvm'>
+        <name>{virtual_name}</name>
+        <memory>2048000</memory>
+        <currentMemory>2048000</currentMemory>
+        <vcpu>2</vcpu>
+        <os>
+        <type arch='x86_64' machine='pc'>hvm</type>
+        <bootmenu enable='no'/>
+        </os>
+        <features><acpi/><apic/><pae/></features>
+        <clock offset='localtime'/>
+        <on_poweroff>destroy</on_poweroff>
+        <on_reboot>restart</on_reboot>
+        <on_crash>destroy</on_crash>
+        <devices>
+        <emulator>/usr/bin/kvm-spice</emulator>
+        <disk type='file' device='disk'>
+        <driver name='qemu' type='qcow2'/>
+        <source file='/opt/{v_name_for_source}.qcow2' span="qcow2"/>
+        <target dev='hda' bus='ide'/>
+        <boot order='1'/>
+        </disk>
+        <disk type='file' device='cdrom'>
+        <source file='/root/rancheros-v1.4.2-rc1.iso'/>
```

Comment:
Using this value `/root/rancheros-v1.4.2-rc1.iso` is really a bad idea.
This `/opt/os-tests/rancheros.iso` may be better.